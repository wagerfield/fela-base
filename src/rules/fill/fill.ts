import { ALL } from "../../core/edges"
import { Edge } from "../../types"
import edgeRule from "../edge/edge"

export interface FillRuleOptions {
  position?: string
  value?: any
  edge?: Edge
}

export default ({
  position = "absolute",
  value = 0,
  edge = ALL
}: FillRuleOptions = {}) => ({
  position,
  ...edgeRule({ edge, value })
})
