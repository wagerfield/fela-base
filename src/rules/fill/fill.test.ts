import fill from "./fill"

describe("fill(props)", () => {
  it("renders expected snapshot with no props", () => {
    expect(fill()).toMatchSnapshot()
  })

  it("renders expected snapshot with position", () => {
    expect(
      fill({
        position: "fixed"
      })
    ).toMatchSnapshot()
  })

  it("renders expected snapshot with edge", () => {
    expect(
      fill({
        edge: "top"
      })
    ).toMatchSnapshot()
  })
})
