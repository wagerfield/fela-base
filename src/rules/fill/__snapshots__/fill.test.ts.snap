// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fill(props) renders expected snapshot with edge 1`] = `
Object {
  "position": "absolute",
  "top": 0,
}
`;

exports[`fill(props) renders expected snapshot with no props 1`] = `
Object {
  "bottom": 0,
  "left": 0,
  "position": "absolute",
  "right": 0,
  "top": 0,
}
`;

exports[`fill(props) renders expected snapshot with position 1`] = `
Object {
  "bottom": 0,
  "left": 0,
  "position": "fixed",
  "right": 0,
  "top": 0,
}
`;
